File: Day6/Complement of Base 10 Integer/1009_2.cs
public class Solution 
{
    public int BitwiseComplement(int n) 
    {
        if(n == 0)
        {
            return 1;
        }
        
        var m = n;
        var mask = 0;
        
        while(m != 0)
        {
            mask = (mask << 1) | 1;
            m = m >> 1;
        }
        
        var ans = (~n) & mask;
        
        return ans;
    }
}

File: Day6/Complement of Base 10 Integer/1009_2.py
class Solution:
    def bitwiseComplement(self, n: int) -> int:
        if n == 0:
            return 1
        
        mask = 0
        m = n
        
        while m != 0:
            mask = (mask << 1) | 1
            m = m >> 1
            
        ans = (~n) & mask
        
        return ans
        